### PR TITLE
Allow Multiple Scans To Be Published

### PR DESCRIPTION
**Pull request by vanderby**
This will allow multiple scans to be published to the build summary. A use case would be an Angular project with .NET backend where one scan would come from MSBUILD and the other from the sonar-scanner on the TypeScript front end.

I also added placing the project name next to the quality gate to differentiate which report belongs to which gate.

image

### CODE REVIEW

**Comment by henryju** [comment]
Sound interesting. @gregaubert can you have a look when you have time?

File: common/ts/publish-task.ts [hunk 1]
```diff
@@ -22,28 +22,36 @@ export default async function publishTask(endpointType: EndpointType) {
...
+        } else {
+          throw e;
+        }
+      }
+    });
+  });
+
+  const buildSummaries = await Promise.all(buildSummariesPromises);
+  const reducer = (accumulator: string, currentValue: string) => `${accumulator}\r\n${currentValue}`;
+  publishBuildSummary(buildSummaries.reduce(reducer), endpoint.type);
```
Comment (gregaubert) [inline common/ts/publish-task.ts]: And with the previous change, you don't need that reduce function and you can just use `Array.join` on the analyses:
```ts
publishBuildSummary(analyses.join('\r\n'), endpoint.type);
```

File: common/ts/sonarqube/Analysis.ts [hunk 2]
```diff
@@ -62,7 +63,7 @@ export default class Analysis {
       font-size: 12px;
       margin-left: 15px;`;
     return `<div style="padding-top: 8px;">
-      <span>Quality Gate</span>
+      <span>${this.projectName} Quality Gate</span>
```
Comment (gregaubert) [inline common/ts/sonarqube/Analysis.ts]: We don't want to add the project name if there is only one analysis to be displayed.

**Comment by vanderby** [comment]
Updated per the PR comments.

**Comment by gregaubert** [comment]
Thanks for the update! You still need to rebase on feature/greg/fix_build so we can have that green build and tests running.

**Comment by gregaubert** [comment]
Indeed there is another failure due to the PR being external. I cherry picked your changes and created a new internal PR to run the analysis. I'll do a few more tests when I have time and then merge it.

**Comment by gregaubert** [comment]
Changes have been merged from the other PR, so I'll close this one. Thanks for your contribution! :+1:

**Pull request closed without merging**